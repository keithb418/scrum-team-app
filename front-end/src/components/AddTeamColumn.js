import React from "react";
import { Button } from "react-bootstrap";
import FontAwesome from "react-fontawesome";

const AddTeamColumn = () => {
  return (
    <Button className="add-team-button">
      <span className="content">
        <FontAwesome
          name="plus-circle"
        />
        Add Team Column
      </span>
    </Button>
  );
};

export default AddTeamColumn;
