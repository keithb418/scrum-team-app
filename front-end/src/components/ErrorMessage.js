import React from "react";
import { Alert } from "react-bootstrap";

const ErrorMessage = ({ message }) =>
  message
    ? <div><Alert bsStyle="warning"><strong>Error:</strong> {message}</Alert></div>
    : null;

export default ErrorMessage;