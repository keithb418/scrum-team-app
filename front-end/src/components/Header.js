import React from "react";
import { Row, Col } from "react-bootstrap";

const Header = () => {
  return (
    <header>
      <Row>
        <Col xs={12}>
          <div className="page-header">
            <h1>Scrum Team</h1>
          </div>
        </Col>
      </Row>
    </header>
  );
};

export default Header;
