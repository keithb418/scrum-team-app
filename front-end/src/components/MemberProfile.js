import React, { Component } from "react";
import { connect } from "react-redux";
import { Button, Row, Col, Panel } from "react-bootstrap";
import { fetchProfileData } from "../actions/shared";
import FontAwesome from "react-fontawesome";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { capitalizeFirstLetter } from "../utils/stringHelpers";
import Loading from "./Loading";
import Profile from "./Profile";
import Skills from "./Skills";

export class MemberProfile extends Component {
  constructor () {
    super();
  }
 
  componentDidMount () {
    this.props.fetchProfileData(this.props.match.params.id);
  }
 
  render () {
    const { teamMember, team } = this.props;
    return !teamMember ?
        <Loading /> :
        <Row>
          <Col className="profile-section" md={6} mdOffset={3} xs={12}>
            <Panel.Heading>
              <h3 className="panel-title">{team.name}</h3>
            </Panel.Heading>
            <hr />
            <div className="member-profile-subheading">
              <FontAwesome name="user-circle" className="member-img" />
              <p>{teamMember.name}</p>
            </div>
            <hr />
            <Row>
              <Profile 
                member={teamMember}/>
              <Skills 
                member={teamMember}/>
            </Row>
            <hr />
            <Link to={`/member/${teamMember._id}/edit`}>
              <Button bsStyle="primary">Edit</Button>
            </Link>
            <Link to="/">
              <Button>Back to Dashboard</Button>
            </Link>
          </Col>
        </Row>;
  }
};

MemberProfile.defaultProps = {
  team: {
    name: ""
  },
  isFetching: true
};

MemberProfile.propTypes = {
  teamMember: PropTypes.object,
  team: PropTypes.object
};

const mapStateToProps = ({ teamMembers: { teamMember }, teams: { teams } }) => {
  if(teamMember) {
    const team = teams.find(team => {
      return team._id === teamMember.team;
    });
    return {
      team,
      teamMember
    };
  } else {
    return {
      teamMember
    };
  }  
};

const mapDispatchToProps = (dispatch) => ({
  fetchProfileData: (id) => dispatch(fetchProfileData(id))
});


export default connect(mapStateToProps, mapDispatchToProps)(MemberProfile);