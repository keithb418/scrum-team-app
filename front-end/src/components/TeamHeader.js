import React, { Component } from "react";
import { Button, Row, Col } from "react-bootstrap";
import { connect } from "react-redux";
import FontAwesome from "react-fontawesome";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { truncateString } from "../utils/stringHelpers";
import { handleDeleteTeam } from "../actions/teams";

export class TeamHeader extends Component {
  constructor () {
    super()
  }

  handleDelete = () => {
    this.props.deleteTeam(this.props.id);
  }

  render () {
    const { teamName, id, teamMembers } = this.props;
    const isDisabled = teamMembers.length > 0;
    return (
      <div className="team-header panel-heading">
        <Link to={`team/${id}/member/add`}>
          <Button className="add-team-member-btn">
            <FontAwesome name="user-plus" />
          </Button>
        </Link>
          <h2 className="col-xs-10 panel-title">{truncateString(teamName, 16)}</h2>
        <Link to={`team/${id}/edit`}>
          <Button>
            <FontAwesome name="edit" />
          </Button>
        </Link>
        <Button
          disabled={isDisabled}
          onClick={this.handleDelete}>
          <FontAwesome name="trash" />
        </Button>
      </div>
    );
  }
};

const mapDispatchToProps = dispatch => ({
  deleteTeam: (id) => dispatch(handleDeleteTeam(id))
})

TeamHeader.propTypes = {
  teamName: PropTypes.string,
  id: PropTypes.string
};

export default connect(undefined, mapDispatchToProps)(TeamHeader);