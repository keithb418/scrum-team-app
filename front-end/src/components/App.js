import React from "react";
import TeamColumnsContainer from "../containers/TeamColumnsContainer";
import AddTeamMember from "./forms/AddTeamMember";
import Header from "./Header";

const App = () => (
  <div>
    <Header />
    <TeamColumnsContainer />
    <AddTeamMember />
  </div>
);

export default App;
