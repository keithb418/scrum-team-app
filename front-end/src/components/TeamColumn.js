import React from "react";
import { connect } from "react-redux";
import TeamMember from "./TeamMember";
import TeamHeader from "./TeamHeader";

let TeamColumn = ({ id = "", teamName = "" , project = "", teamMembers = [], dispatch }) => {

  teamMembers.sort((a, b) => {
    if (b.teamLead) {
      return 1;
    }

    if (a.teamLead) {
      return -1;
    }

    return 0;
  });

  const teamMemberComponents = teamMembers.map((teamMember) =>
    <TeamMember 
      key={teamMember._id}
      id={teamMember._id} 
      name={teamMember.name} 
      teamLead={teamMember.teamLead} 
      role={teamMember.role} />
  );

  let allowDrop = (e) => {
    e.preventDefault();
  };

  let drop = (e) => {
    e.preventDefault();

    dispatch({
      type: "CHANGE_TEAM",
      teamMemberId: e.dataTransfer.getData("tmId"),
      team: id
    });
  };

  return (
    <div className="panel panel-default team-column" onDragOver={allowDrop} onDrop={drop}>
      <TeamHeader 
        id={id}
        teamName={teamName} 
        project={project} />
      <div className="panel-body team-body" onDrop={drop} onDragOver={allowDrop} >
        {teamMemberComponents}
      </div>
    </div>
  );
};

TeamColumn = connect()(TeamColumn);

export default TeamColumn;