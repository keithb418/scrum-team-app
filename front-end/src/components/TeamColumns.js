import React from "react";
import TeamColumn from "./TeamColumn";
import AddTeamColumn from "./AddTeamColumn";

const TeamColumns = ({ teams }) => {
  
  let teamColumns = teams.map((team) => 
    <TeamColumn
      key={team._id}
      id={team._id} 
      teamName={team.name} 
      project={team.project}
      teamMembers={team.teamMembers} />
  );

  return (
    <div className="team-columns">
      {teamColumns}
      <AddTeamColumn />
    </div>
  );
};

export default TeamColumns;