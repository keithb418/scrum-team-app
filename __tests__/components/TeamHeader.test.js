import React from 'react';
import { shallow } from 'enzyme';
import { TeamHeader } from "../../front-end/src/components/TeamHeader";
import teamMembersData from '../fixtures/teamMembers';
import teamsData from '../fixtures/teams';
let wrapper, button, deleteTeam, truncateString;

beforeEach(() => {
    truncateString = jest.fn();
    deleteTeam = jest.fn();
    wrapper = shallow(       
        <TeamHeader 
          teamMembers={teamMembersData}
          id={teamsData[0]._id}
          deleteTeam={deleteTeam}
          teamName={teamsData[0].name}/>
    );
    button = wrapper.find('Button').at(2);
});

test('should render the TeamHeader component correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(button.props().disabled).toBe(true);
});

test('should enable delete button if team members are empty', () => {
    wrapper = shallow(
        <TeamHeader 
        teamMembers={[]}
        deleteTeam={deleteTeam}
        id={teamsData[0]._id}
        teamName={teamsData[0].name}/>
    );
    button = wrapper.find('Button').at(2);
    expect(button.props().disabled).toBe(false);
});

test('should handle deleting Team', () => {
  button.simulate('click')
  expect(deleteTeam).toHaveBeenCalledWith(teamsData[0]._id);
});

