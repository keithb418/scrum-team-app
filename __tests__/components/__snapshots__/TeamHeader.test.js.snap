// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the TeamHeader component correctly 1`] = `
<div
  className="team-header panel-heading"
>
  <Link
    replace={false}
    to="team/1/member/add"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="add-team-member-btn"
      disabled={false}
    >
      <FontAwesome
        name="user-plus"
      />
    </Button>
  </Link>
  <h2
    className="col-xs-10 panel-title"
  >
    Team 1
  </h2>
  <Link
    replace={false}
    to="team/1/edit"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
    >
      <FontAwesome
        name="edit"
      />
    </Button>
  </Link>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <FontAwesome
      name="trash"
    />
  </Button>
</div>
`;
